### PR TITLE
Make forms responsive and make errors display per field

### PR DESCRIPTION
**Pull request by tiffz**
This PR is to fix some usability issues that have been bothering me with forms. 

Here's a screenshot that shows both the responsiveness and also what errors look like now. Both server side and clientside errors will look like this after this change. If this is acceptable, I will adjust all forms on the site to match login/registration. 
screenshot 2017-03-20 01 41 29

Non-responsively, forms look relatively the same. Some minor spacing adjustments and such. 

Also, this adjusts forms to automatically do clientside validation on blur, which includes removing any server side errors immediately after you fix them (once you unfocus).

### CODE REVIEW

File: saylua/modules/users/templates/login/register.html [hunk 1]
```diff
@@ -1,60 +1,34 @@
...
-        <small class="form-tip">Valid characters: letters, numbers, and +~._-</small>
-      </td>
-    </tr>
-    <tr>
-      <th class="label">
-        {{ form.password.label }}
-      </th>
-      <td>
-        {{ form.password() }}
-        <small class="form-tip">Passwords must be at least {{config['MIN_PASSWORD_LENGTH']}} characters. </small>
```
Comment (NoiSek) [inline saylua/modules/users/templates/login/register.html]: It's gone now, but I wonder if we could add a linting rule for this somehow. Our `{{ var }}` vs `{{var}}` usage throughout the codebase is somewhat inconsistent.

**Review by NoiSek — APPROVED** [review]
Perhaps you should consider increasing the required password complexity / length while you're here. Looks good otherwise.

(But note that you have linter errors)

File: saylua/modules/users/templates/login/register.html (hunk 1 above)
Comment (tiffz) [inline saylua/modules/users/templates/login/register.html]: That would be a good idea, yeah. Do you think it'd be better to do {{ var }} or {{var}}?

Comment (NoiSek) [inline saylua/modules/users/templates/login/register.html]: I personally prefer `{{ var }}`, it's the more common convention for readability.

**Pull request merged**